### ISSUE TITLE
Support for using the same name label in the downloaded files

### ISSUE THREAD

**Issue by skrishn** [Esri/solutions-components#160]
There is an option to customize the label for notification, but the downloads have a different name. If the downloads have the same name that the user provided it would be beneficial for users.

**Comment by MikeTschudi** [comment]
May I please have more details about this request?

**Comment by skrishn** [comment]
@MikeTschudi I missed that scenario and thank you for catching it. 

**For Address search:**
1. If the input is 'Naperville, Illinois', I notice the notification label comes as 'Naperville, Illinois 100 meters'.
2. If the input is a full address (like yours), it comes as 123 Water St, Naperville, Illinois, 60540 100 meters

I feel that using 'Naperville, 100 meters' or '123 Water St, Naperville, 100 meters' is intuitive and simple.

**For sketch:**
"Sketch 1 kilometers" is good. 


I also noticed a behavior that appears to be a bug that could affect this implementation. 

1. Create a notification
2. Click edit on the notification

![image](https://user-images.githubusercontent.com/3836616/218183776-adef59ee-d8ab-44f3-b5aa-5020acd2adb4.png)
3. Edit the address on the address bar. Notice the notification label doesn't update.
![image](https://user-images.githubusercontent.com/3836616/218184162-df816686-f623-46f5-ac4f-cde9027dc503.png)
4. Hit done
5. Notice the notification label is updated with the current address
6. 
![image](https://user-images.githubusercontent.com/3836616/218184335-baa83504-2116-4e95-940f-871fbebfb824.png)

**Comment by aparveen** [comment]
@skrishn label updates after you click Done when you edit your search criteria seems an expected behavior to me, why do you think it is a bug and what is your expected behavior

**Comment by skrishn** [comment]
@aparveen I agree. I was thinking the label gets updated dynamically, and since we discussed that it doesn't, I think we're good.

**Comment by aparveen** [comment]
Instant App team got the same request in dev summit

**Comment by aparveen** [comment]
label box for custom name, optional, no custom label , default name and think about multiple labels

**Comment by aparveen** [comment]
@MikeTschudi, Can we use individual names of the list when list names are provided,  and when list name is not provided we can use default list name.
For multiple lists we can take advantage of the PDF title the IA team proposed to add and use that as file name if provided, when the title is not provided we can use the current pattern we are using. Please let me know your thoughts on this.

**Comment by MikeTschudi** [comment]
@aparveen, we cannot at this time. The PDF library only receives a list of selection set names and a list of ids, with no mapping between the two.

What I've implemented as a prototype creates filenames such as
* 1 selection set:  "Sketch 2 kilometers.pdf"
* 2 selection sets: "Sketch 1000 feet, Sketch 0 kilometers.pdf"

and internal headings such as (single selection set)
![image](https://user-images.githubusercontent.com/2125181/226698193-8f817bf7-ccba-4509-b66e-0672df647c3d.png)

**Comment by aparveen** [comment]
Is this change already merged to devext and ready for testing?

**Comment by MikeTschudi** [comment]
Not yet--still working through build issues before creating the PR

**Comment by aparveen** [comment]
@MikeTschudi Instant Apps team requested to Add map in the PDF, Kelly mentioned that they will only want the the title on the map page, otherwise title placed on the label page can create problem printing the labels in sticker printing paper

**Comment by MikeTschudi** [comment]
The title is always outside the label area.

**Comment by aparveen** [comment]
@MikeTschudi it looks good,  I really like the way you concatenated the list names together.
Just noticed one issue, if there are multiple files, but for download I checked only one or two, it still shows the full concatenated name, is it possible to check what files are checked for downloading and use their names only

Test App: https://sg-em-devext.mapsdevext.arcgis.com/apps/instant/notification/index.html?appid=005416b95280406a91961ae085922c25&locale=en-US

![image](https://user-images.githubusercontent.com/10148401/230508560-e39a1a9c-1192-4422-b8ce-0dfe4e465143.png)

**Comment by aparveen** [comment]
Verified
